src/modules/twap/services/setupExtensibleFallbackHandler.test.ts

```diff
@@ -0,0 +1,74 @@
+import { setupExtensibleFallbackHandler } from './setupExtensibleFallbackHandler'
+import { renderHook } from '@testing-library/react-hooks'
+import { useUpdateAtom } from 'jotai/utils'
+import { walletInfoAtom } from 'modules/wallet/api/state'
+import { useEffect } from 'react'
+import { useExtensibleFallbackContext } from '../hooks/useExtensibleFallbackContext'
+import { useSafeAppsSdk } from 'modules/wallet/web3-react/hooks/useSafeAppsSdk'
+import { useGP2SettlementContract } from 'legacy/hooks/useContract'
+import { WithMockedWeb3 } from 'test-utils'
+import { useWeb3React } from '@web3-react/core'
+import { Web3Provider } from '@ethersproject/providers'
+
+const chainId = 5
+
+jest.mock('modules/wallet/web3-react/hooks/useSafeAppsSdk')
+jest.mock('legacy/hooks/useContract')
+jest.mock('@web3-react/core', () => {
+  return {
+    ...jest.requireActual('@web3-react/core'),
+    useWeb3React: jest.fn(),
+  }
+})
+
+const defaultJsonRpcHandler = (method: string, params?: any[]) => {
+  if (method === 'eth_chainId') return Promise.resolve(5)
+
+  // domainVerifiers()
+  if (method === 'eth_call' && params?.[0]?.data?.startsWith('0x51cad5ee')) {
+    //Composable cow address
+    return Promise.resolve('0x000000000000000000000000a31b99bd44528c7bae9e1f675d810ae13b0e29aa')
+  }
+
+  return Promise.resolve(null)
+}
+
+const useWeb3ReactMock = useWeb3React as jest.MockedFunction<typeof useWeb3React>
+const useSafeAppsSdkMock = useSafeAppsSdk as jest.MockedFunction<typeof useSafeAppsSdk>
+const useGP2SettlementContractMock = useGP2SettlementContract as jest.MockedFunction<typeof useGP2SettlementContract>
+
+describe('setupExtensibleFallbackHandler - integration test', () => {
```

Comment:
Not really integration, is it?
Everything is mocked, this just assumes the flow goes as intended, but does not execute any transaction to observe the end to end use scenario, no?

Reply:
`integration` on the components level, not the system. In this case, I test integration between `useExtensibleFallbackContext` + `setupExtensibleFallbackHandler`.

I know, usually we understand `integration` tests only on system level, so, I can remove that word just to avoid confusion